prow/plugins/dco/dco.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dco
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"regexp"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName               = "dco"
+	dcoContextName           = "dco"
+	dcoContextMessageFailed  = "A commit in PR is missing Signed-off-by"
+	dcoContextMessageSuccess = "All commits have Signed-off-by"
+
+	dcoYesLabel        = "dco-signoff: yes"
+	dcoNoLabel         = "dco-signoff: no"
+	dcoMsgPruneMatch   = "Thanks for your pull request. Before we can look at your pull request, you'll need to add a 'DCO signoff' to your commits."
+	dcoNotFoundMessage = `Thanks for your pull request. Before we can look at your pull request, you'll need to add a 'DCO signoff' to your commits.
+
+<details>
```

Comment:
This message could probably do with being expanded further?

Reply:
I think it should also link the full DCO statement like that: https://developercertificate.org/